app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_post_counter
+    author.post_counter += 1
+    author.save
+  end
```

Comment:
- [ ] It would be nice to use ``after_save :update_user_post_counter`` what does it do: update_post_counter method will be called automatically executed after the model instance is saved to the database. Also make sure to that `update_user_post_counter` method private.
- [ ] The `after_save` callback is one such callback that gets executed after an object has been successfully saved to the database.